File: JavaScript/boj/Silver/15649.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? '/dev/stdin' : './input.txt';
const [N, M] = fs
  .readFileSync(filePath)
  .toString()
  .trim()
  .split(' ')
  .map(Number);

const list = Array.from({ length: N }, (_, i) => i + 1);

const getPermutation = (arr, selectNum) => {
  const result = [];

  if (selectNum === 1) return arr.map((el) => [el]);

  arr.forEach((fixed, index, origin) => {
    const rest = [...origin.slice(0, index), ...origin.slice(index + 1)];
    const permutation = getPermutation(rest, selectNum - 1);
    const attached = permutation.map((el) => [fixed, ...el]);

    result.push(...attached);
  });

  return result;
};

const permutations = getPermutation(list, M);
permutations.forEach((el) => {
  console.log(el.join(' '));
});

File: Python/boj/Silver/15649.py
import sys
from itertools import permutations

if __name__ == "__main__":
    try:
        sys.stdin = open("input.txt", "r")
    except FileNotFoundError:
        pass

    N, M = map(int, input().split())
    list = [i for i in range(1, N + 1)]

    result = permutations(list, M)

    for i in result:
        print(" ".join(map(str, i)))
